scripts/create-price-discovery-offer.ts

```diff
@@ -0,0 +1,105 @@
+import fs from "fs";
+import { EnvironmentType } from "@bosonprotocol/common/src/types/configs";
+import { providers, Wallet } from "ethers";
+import { program } from "commander";
+import { getEnvConfigById } from "@bosonprotocol/common/src";
+import { CoreSDK, offers } from "../packages/core-sdk/src";
+import { EthersAdapter } from "../packages/ethers-sdk/src";
+
+program
+  .description("Create a Price Discovery Offer.")
+  .argument(
+    "<SELLER_PRIVATE_KEY>",
+    "Private key of the Seller account (assistant role)."
+  )
+  .argument("<OFFER_DATA>", "JSON file with the Offer parameters")
+  .option("-e, --env <ENV_NAME>", "Target environment", "testing")
+  .option("-c, --configId <CONFIG_ID>", "Config id", "testing-80002-0")
+  .parse(process.argv);
+
+async function main() {
+  const [sellerPrivateKey, offerDataJsonFile] = program.args;
+
+  const opts = program.opts();
+  const envName = opts.env || "testing";
+  const configId = opts.configId || "testing-80002-0";
+  const defaultConfig = getEnvConfigById(envName as EnvironmentType, configId);
+  const chainId = defaultConfig.chainId;
+  const rawData = fs.readFileSync(offerDataJsonFile);
+  const offerDataJson = JSON.parse(
+    rawData.toString()
+  ) as offers.CreateOfferArgs;
+  if (offerDataJson["validFromDateInMS"] === undefined) {
+    offerDataJson["validFromDateInMS"] = Date.now();
+  }
+  if (offerDataJson["voucherRedeemableFromDateInMS"] === undefined) {
+    offerDataJson["voucherRedeemableFromDateInMS"] = Date.now();
+  }
+  // const WETH_ADDRESS = "0x52eF3d68BaB452a294342DC3e5f464d7f610f72E"; // Amoy
+  const WETH_ADDRESS = "0x7b79995e5f793a07bc00c21412e50ecae098e7f9"; // Sepolia
```

Comment:
use one or the other depending on configId/envName?

Reply:
Changed. Now reading the WETH address from the coreSDK config